In [162]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

loading the scraped file(csv)

In [163]:
data = pd.read_csv('auto_scraper.csv')

In [164]:
#Lets view the first five rows 

In [165]:
data.head()

Unnamed: 0                                          Car_Title Car_Price  \
0           0          Hyundai Tucson 1.6 GDi SE Manual 5dr        £13,699   
1           1   BMW 2 Series 3.0 M235i Sport Auto (s/s) 2dr        £19,000   
2           2                Nissan Note 1.6 16v Acenta 5dr         £1,205   
3           3            Volkswagen Touareg 5.0 TDI V10 5dr         £5,750   
4           4   Audi A1 2.0 TDI Amplified Edition Sportback 5...   £35,000   

  Car_Year Car_Type_Door Car_Mileage Car_engine_size Car_Horse_PW  \
0     2018           SUV      19,228            1.6L        132PS   
1     2016         Coupe      12,000            3.0L       326BHP   
2     2007     Hatchback      70,699            1.6L       109BHP   
3     2005           SUV     137,000            5.0L       307BHP   
4     2012     Hatchback      95,000            2.0L       141BHP   

  Car_Type_Vehicle Car_Filling_in  
0           Manual         Petrol  
1        Automatic         Petrol  
2           Manual         Petrol  
3        Automatic         Diesel  
4           Manual         Diesel

In [166]:
print('Scraped data:',data.shape)

Scraped data: (9575, 10)


# Missing Values

In [167]:
#Let check if the data has any missing values

In [168]:
data.isnull().sum()

Unnamed: 0            0
Car_Title             0
Car_Price             0
Car_Year            384
Car_Type_Door       384
Car_Mileage         384
Car_engine_size     384
Car_Horse_PW        384
Car_Type_Vehicle    384
Car_Filling_in      384
dtype: int64

- There are 384 Missing data across the dataFrame and I will be dropping these rows.
- I will also be dropping 'Unnamed: 0' columns as it provides no helpful information.
- Further I will investigate if all the columns had the right data scrapered.

In [169]:
data = data.dropna()

In [170]:
data = data.drop('Unnamed: 0',axis=1)

In [171]:
data.Car_Mileage.value_counts()

Petrol    176
Diesel     92
Van        57
60,000     47
80,000     47
         ... 
1,053       1
3,749       1
20,590      1
6,855       1
75,525      1
Name: Car_Mileage, Length: 6104, dtype: int64

In [172]:
data.Car_Filling_in.value_counts()

Petrol          4467
Diesel          3876
included         268
Hybrid           211
miles            111
Manual            39
1                 39
Full              29
ULEZ              20
warranty          19
Automatic         17
2                 16
year              15
owners            13
3                  8
manufacturer       7
tax                6
service            5
owner              5
Part               4
4                  4
–                  3
road               2
Bi                 1
94BHP              1
125PS              1
74BHP              1
67BHP              1
110BHP             1
5                  1
Name: Car_Filling_in, dtype: int64

It look we have scrapered some data that imported into wrong columns. So I'm going to filter the data for Petrol, Diesel and Hybrid cars only.


In [173]:
data = data[data['Car_Filling_in'].isin(['Petrol','Diesel','Hybrid'])]

In [174]:
data = data[data['Car_Mileage'] != 'Van']

In [175]:
print('After Removing the missing/wrong data:',data.shape[0])

After Removing the missing/wrong data: 8553


We Can extract the make of the car from Car_Title Column 

In [176]:
data['Car_Make'] = data.Car_Title.str.split().str.get(0)

In [177]:
data.head()

Car_Title Car_Price Car_Year  \
0          Hyundai Tucson 1.6 GDi SE Manual 5dr        £13,699     2018   
1   BMW 2 Series 3.0 M235i Sport Auto (s/s) 2dr        £19,000     2016   
2                Nissan Note 1.6 16v Acenta 5dr         £1,205     2007   
3            Volkswagen Touareg 5.0 TDI V10 5dr         £5,750     2005   
4   Audi A1 2.0 TDI Amplified Edition Sportback 5...   £35,000     2012   

  Car_Type_Door Car_Mileage Car_engine_size Car_Horse_PW Car_Type_Vehicle  \
0           SUV      19,228            1.6L        132PS           Manual   
1         Coupe      12,000            3.0L       326BHP        Automatic   
2     Hatchback      70,699            1.6L       109BHP           Manual   
3           SUV     137,000            5.0L       307BHP        Automatic   
4     Hatchback      95,000            2.0L       141BHP           Manual   

  Car_Filling_in    Car_Make  
0         Petrol     Hyundai  
1         Petrol         BMW  
2         Petrol      Nissan  
3         Diesel  Volkswagen  
4         Diesel        Audi

# Changing the Dtypes 

In [178]:
data.dtypes

Car_Title           object
Car_Price           object
Car_Year            object
Car_Type_Door       object
Car_Mileage         object
Car_engine_size     object
Car_Horse_PW        object
Car_Type_Vehicle    object
Car_Filling_in      object
Car_Make            object
dtype: object

- Converting Car_Price and Car_Milleage into Numeric Values
- Before that remove £ sign and comma from each of the columns

In [179]:
data['Car_Price'] = data['Car_Price'].str.replace('£','')
data['Car_Price'] = data['Car_Price'].str.replace(',','')

In [182]:
data['Car_Price'] = pd.to_numeric(data['Car_Price'])

In [183]:
data['Car_Mileage'] = data['Car_Mileage'].str.replace(',','')

In [185]:
data['Car_Mileage'] = pd.to_numeric( data['Car_Mileage'])

In [186]:
data.head()

Car_Title  Car_Price Car_Year  \
0          Hyundai Tucson 1.6 GDi SE Manual 5dr           13699     2018   
1   BMW 2 Series 3.0 M235i Sport Auto (s/s) 2dr           19000     2016   
2                Nissan Note 1.6 16v Acenta 5dr            1205     2007   
3            Volkswagen Touareg 5.0 TDI V10 5dr            5750     2005   
4   Audi A1 2.0 TDI Amplified Edition Sportback 5...      35000     2012   

  Car_Type_Door  Car_Mileage Car_engine_size Car_Horse_PW Car_Type_Vehicle  \
0           SUV        19228            1.6L        132PS           Manual   
1         Coupe        12000            3.0L       326BHP        Automatic   
2     Hatchback        70699            1.6L       109BHP           Manual   
3           SUV       137000            5.0L       307BHP        Automatic   
4     Hatchback        95000            2.0L       141BHP           Manual   

  Car_Filling_in    Car_Make  
0         Petrol     Hyundai  
1         Petrol         BMW  
2         Petrol      Nissan  
3         Diesel  Volkswagen  
4         Diesel        Audi

# export to CSV

In [187]:
data.to_csv('Data_Cleaning.csv')